@xen-orchestra/lite/src/components/ui/UiToggle.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <label class="toggle">
+    <input
+      type="checkbox"
+      :checked="props.value"
+      :disabled="props.disabled"
+      @change="onChange"
+    />
+    <span class="slider"></span>
+  </label>
+</template>
+
+<script lang="ts" setup>
+const props = withDefaults(
+  defineProps<{
+    disabled?: boolean;
+    onChange?: (value: boolean) => void;
+    value?: boolean;
+    scale?: number;
+  }>(),
+  {
+    disabled: false,
+    value: false,
+    scale: 1,
+  }
+);
+
+const onChange = (event: Event) => {
+  if (props.onChange !== undefined) {
+    props.onChange((event.target as HTMLInputElement).checked);
+  }
+};
```

Comment:
We should use a `modelValue` prop here, this allow to use directly `v-model` on the component (`<UiToggle v-model="acceptAgreement" />`)

And with `useVModel()` we don't have to handle value and event ourselves :)

```suggestion
import { useVModel } from "@vueuse/core";

const props = withDefaults(
  defineProps<{
    modelValue?: boolean;
    disabled?: boolean;
    scale?: number;
  }>(),
  { scale: 1 }
);

const emit = defineEmits<{
  (event: "update:modelValue", value: boolean): void;
}>();

const value = useVModel(props, "modelValue", emit);
```